workers/unity/Packages/com.improbable.gdk.core/Subscriptions/StandardSubscriptionManagers/WorkerFlagSubscriptionManager.cs

```diff
@@ -0,0 +1,102 @@
+using System;
+using System.Collections.Generic;
+using Improbable.Gdk.Subscriptions;
+using Unity.Entities;
+
+namespace Improbable.Gdk.Core
+{
+    [AutoRegisterSubscriptionManager]
+    public class WorkerFlagSubscriptionManager : SubscriptionManager<WorkerFlagReader>
+    {
+        private World world;
+
+        private HashSet<Subscription<WorkerFlagReader>> readerSubscriptions = new HashSet<Subscription<WorkerFlagReader>>();
+
+        public WorkerFlagSubscriptionManager(World world)
+        {
+            this.world = world;
+        }
+
+        public override void Cancel(ISubscription subscription)
+        {
+            var readerSubscription = (Subscription<WorkerFlagReader>) subscription;
+
+            readerSubscription.Value.IsValid = false;
+            readerSubscription.Value.RemoveAllCallbacks();
+            readerSubscriptions.Remove(readerSubscription);
+        }
+
+        public override void ResetValue(ISubscription subscription)
+        {
+            var readerSubscription = (Subscription<WorkerFlagReader>) subscription;
+
+            readerSubscription.Value.RemoveAllCallbacks();
+        }
+
+        public override Subscription<WorkerFlagReader> Subscribe(EntityId entityId)
+        {
+            var subscription = new Subscription<WorkerFlagReader>(this, new EntityId(0));
+            readerSubscriptions.Add(subscription);
+
+            subscription.SetAvailable(new WorkerFlagReader(world));
+
+            return subscription;
+        }
+    }
+
+    public class WorkerFlagReader
+    {
+        public bool IsValid;
```

Comment:
This is never read

Reply:
Ah woops - the callbacks should be wrapped.